### ISSUE TITLE
API Gateway removal failure

### ISSUE THREAD

**Issue by denis256** [gruntwork-io/cloud-nuke#370]
Noticed that `cloud-nuke` fail to remove API Gateway, it fails with error:
```
[cloud-nuke] ERRO[2022-10-27T17:39:57Z] [Failed] Error deleting API Gateway (v2) XYZ in us-east-1 
[cloud-nuke] ERRO[2022-10-27T17:39:57Z] [Failed] BadRequestException: Deleting API XYZ failed. Please remove all API mappings for the API from your custom domain names. 
ERROR: 1 error occurred:
	* BadRequestException: Deleting API XYZ failed. Please remove all API mappings for the API from your custom domain names.
```

**Comment by hongil0316** [comment]
Here's the process I followed to reproduce the issue:

1. Created an HTTP API in API Gateway.
2. Added an endpoint named 'sample.'
3. Connected the 'sample' endpoint to a Lambda function, which simply returns 'hello from lambda' as a response.
4. Added a custom domain ([[api.xxx.com](http://api.xxx.com/)](http://api.xxx.com/)).
5. In 'API mappings,' I mapped the HTTP API's dev stage.

![image](https://github.com/gruntwork-io/cloud-nuke/assets/96548424/4378e9f3-fa75-415f-8206-2ece840b5792)

You might have an old version of the application.
